crates/pera-core/src/dwallet_mpc/mpc_manager.rs

```diff
@@ -189,39 +219,67 @@ impl DWalletMPCManager {
 
     fn handle_event(&mut self, event: Event, session_info: SessionInfo) -> DwalletMPCResult<()> {
         self.outputs_verifier.handle_new_event(&session_info);
-        if let Ok((party, auxiliary_input, session_info)) = from_event(
-            &event,
-            &self,
-            authority_name_to_party_id(&self.epoch_store()?.name, &*self.epoch_store()?)?,
-        ) {
-            self.push_new_mpc_session(auxiliary_input, party, session_info)?;
-        };
+        let (public_input, private_input) = session_input_from_event(&event, &self)?;
+        self.push_new_mpc_session(public_input, private_input, session_info)?;
         Ok(())
     }
 
-    // todo(zeev): doc this.
-    pub fn get_decryption_share(&self) -> DwalletMPCResult<DecryptionKeyShare> {
+    pub(super) fn get_protocol_public_parameters(
+        &self,
+        key_scheme: DWalletMPCNetworkKeyScheme,
+        key_version: u8,
+    ) -> DwalletMPCResult<Vec<u8>> {
+        if let Some(self_decryption_share) = self.epoch_store()?.dwallet_mpc_network_keys.get() {
+            return self_decryption_share.get_protocol_public_parameters(key_scheme, key_version);
+        }
+        Err(DwalletMPCError::TwoPCMPCError(
+            "Decryption share not found".to_string(),
+        ))
+    }
+
+    pub(super) fn get_decryption_public_parameters(
+        &self,
+        key_scheme: DWalletMPCNetworkKeyScheme,
+        key_version: u8,
+    ) -> DwalletMPCResult<Vec<u8>> {
+        if let Some(self_decryption_share) = self.epoch_store()?.dwallet_mpc_network_keys.get() {
+            return self_decryption_share.get_decryption_public_parameters(key_scheme, key_version);
+        }
+        Err(DwalletMPCError::TwoPCMPCError(
+            "Decryption share not found".to_string(),
+        ))
+    }
+
+    /// Retrieves the decryption share for the current authority.
+    ///
+    /// This function accesses the current epoch's store and determines the party ID for the
+    /// authority using its name.
+    /// It then retrieves the corresponding decryption share from
+    /// the node configuration.
+    /// The decryption share is combined with the public parameters
+    /// to build a [`DecryptionKeyShare`].
+    /// If any required data is missing or invalid, an
+    /// appropriate error is returned.
+    fn get_decryption_share(
```

Comment:
```suggestion
    fn get_decryption_key_shares(
```
`decryption_share` means something else entirely